sentry-jul/src/main/java/io/sentry/jul/SentryHandler.java

```diff
@@ -0,0 +1,283 @@
+package io.sentry.jul;
+
+import io.sentry.Breadcrumb;
+import io.sentry.Sentry;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.logback.BuildConfig;
+import io.sentry.protocol.Message;
+import io.sentry.protocol.SdkVersion;
+import io.sentry.util.CollectionUtils;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.ErrorManager;
+import java.util.logging.Filter;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.LogRecord;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+import org.slf4j.MDC;
+
+/** Logging handler in charge of sending the java.util.logging records to a Sentry server. */
+public final class SentryHandler extends Handler {
+  /** Name of the {@link SentryEvent} extra property containing the Thread id. */
+  public static final String THREAD_ID = "thread_id";
+  /**
+   * If true, <code>String.format()</code> is used to render parameterized log messages instead of
+   * <code>MessageFormat.format()</code>; Defaults to false.
+   */
+  private boolean printfStyle;
+
+  private @NotNull Level minimumBreadcrumbLevel = Level.INFO;
+  private @NotNull Level minimumEventLevel = Level.SEVERE;
+
+  /** Creates an instance of SentryHandler. */
+  public SentryHandler() {
+    this(new SentryOptions(), true);
+  }
+
+  /** Creates an instance of SentryHandler. */
+  @TestOnly
+  SentryHandler(final @NotNull SentryOptions options, final boolean configureFromLogManager) {
+    setFilter(new DropSentryFilter());
+    if (configureFromLogManager) {
+      retrieveProperties();
+    }
+    if (!Sentry.isEnabled()) {
+      options.setEnableExternalConfiguration(true);
+      options.setSdkVersion(createSdkVersion(options));
+      Sentry.init(options);
+    }
+  }
+
+  @Override
+  public void publish(final @NotNull LogRecord record) {
+    // Do not log the event if the current thread is managed by sentry
+    if (!isLoggable(record)) {
+      return;
+    }
+    try {
+      if (record.getLevel().intValue() >= minimumEventLevel.intValue()) {
+        Sentry.captureEvent(createEvent(record));
+      }
+      if (record.getLevel().intValue() >= minimumBreadcrumbLevel.intValue()) {
+        Sentry.addBreadcrumb(createBreadcrumb(record));
```

Comment:
can we use the `HubAdapter` as a package-private ctor so we can assert this via the mocked hub?

Reply:
oh just saw you've mocked the `transport` instead